### PR TITLE
refactor: decompose TransactionConfirmationModal

### LINKED ISSUE #2429: Create a generic component TradeConfirmationModal (to replace LimitOrdersConfirmModal)
image

For Advanced orders, we will need a component for the confirmation modal.
Need to refactor `LimitOrdersConfirmModal` to create a new universal solution for confirming any trade.

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2429

Just decomposed the legacy component `TransactionConfirmationModal` and removed `mod` file.

[Reference](https://github.com/cowprotocol/cowswap/pull/2569#discussion_r1209970534)

### CODE REVIEW

File: src/legacy/components/TransactionConfirmationModal/LegacyConfirmationPendingContent.tsx [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+function getSubmittedMessage(operationLabel: string, operationType: ConfirmOperationType): string {
+  switch (operationType) {
+    case ConfirmOperationType.ORDER_SIGN:
+      return t`The order is submitted and ready to be settled.`
+    default:
+      return `The ${operationLabel} is submitted.`
+  }
+}
+
+function getWalletNameLabel(walletType: WalletType): string {
```
Comment (nenadV91) [inline src/legacy/components/TransactionConfirmationModal/LegacyConfirmationPendingContent.tsx:97]: Should this be here or in wallets module?

File: src/legacy/components/TransactionConfirmationModal/LegacyConfirmationPendingContent.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+  operationType: ConfirmOperationType
+  chainId: number
+}) {
+  const { connector } = useWeb3React()
+  const { account } = useWalletInfo()
+  const walletDetails = useWalletDetails()
+  const { ensName, isSmartContractWallet } = walletDetails
+  const gnosisSafeInfo = useGnosisSafeInfo()
+
+  const walletType = useMemo((): WalletType => {
```
Comment (nenadV91) [inline src/legacy/components/TransactionConfirmationModal/LegacyConfirmationPendingContent.tsx:126]: This could be a separate hook

Comment (shoom3301) [reply src/legacy/components/TransactionConfirmationModal/LegacyConfirmationPendingContent.tsx:126]: Agree, I would love to refactor it after TWAP launch.
Anyway, in this PR I've not changed the code, just split it into separate files.

File: src/legacy/components/TransactionConfirmationModal/LegacyConfirmationPendingContent.tsx (hunk 1 above)
Comment (shoom3301) [reply src/legacy/components/TransactionConfirmationModal/LegacyConfirmationPendingContent.tsx:97]: Same as here https://github.com/cowprotocol/cowswap/pull/2571#discussion_r1211653921

File: src/legacy/components/TransactionConfirmationModal/LegacyConfirmationModalContent.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  topContent,
+}: ConfirmationModalContentProps) {
+  return (
+    <ContentWrapper className={className}>
+      <Section>
+        <GPModalHeader>
+          <Text fontWeight={600} fontSize={titleSize || 16} style={styles}>
+            {title}
+          </Text>
+          <CloseIconWrapper onClick={() => onDismiss()} />
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply src/legacy/components/TransactionConfirmationModal/LegacyConfirmationModalContent.tsx:33]: Same as here https://github.com/cowprotocol/cowswap/pull/2571#discussion_r1211653921

**Review by nenadV91 — APPROVED** [review]
Left some comments

File: src/legacy/hooks/useApproveCallback/useApproveCallbackMod.ts
Comment (alfetopito) [inline src/legacy/hooks/useApproveCallback/useApproveCallbackMod.ts:1]: Not for this PR, but another candidate to be removed 😅

**Review by alfetopito — APPROVED** [review]

**Pull request merged**